addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -112,6 +114,15 @@ function EntityFactory() {
             return entity;
         }
 
+        /**
+         * @static
+         * @return {Object[]}
+         */
+        static get relations() {
```

Comment:
Ideally this shouldn't be needed/shouldn't exist.

All fields should be iterated the same everywhere, and each should have method doing stuff according to its type (ex. fields that are relational should handle the `isCausal` stuff on a "delete" of them.